Test/dafny0/NoTypeArgs.dfy

```diff
@@ -63,25 +63,23 @@ function reverse(xs: List): List
   case Cons(t, rest) => concat(reverse(rest), Cons(t, Nil))
 }
 
-ghost method Theorem(xs: List)
-  ensures reverse(reverse(xs)) == xs;
+lemma Theorem(xs: List)
+  ensures reverse(reverse(xs)) == xs
 {
-  match (xs) {
-    case Nil =>
-    case Cons(t, rest) =>
-      Lemma(reverse(rest), Cons(t, Nil));
-  }
+  match xs
+  case Nil =>
+  case Cons(t, rest) =>
+    Lemma(reverse(rest), Cons(t, Nil));
 }
 
-ghost method Lemma<A>(xs: List, ys: List)
-  ensures reverse(concat(xs, ys)) == concat(reverse(ys), reverse(xs));
+lemma Lemma<A>(xs: List, ys: List)
+  ensures reverse(concat(xs, ys)) == concat(reverse(ys), reverse(xs))
 {
-  match (xs) {
-    case Nil =>
-      assert forall ws :: concat(ws, Nil) == var ws : List<A> := ws; ws;
-    case Cons(t, rest) =>
-      assert forall a, b, c :: concat(a, concat(b, c)) == var ws : List <A> := concat(concat(a, b), c); ws;
-  }
+  match xs
+  case Nil =>
+    assert forall ws: List<A> :: concat(ws, Nil) == ws;
+  case Cons(t, rest) =>
+    assert forall a: List<A>, b, c :: concat(a, concat(b, c)) == concat(concat(a, b), c);
```

Comment:
I'd prefer the cases be indented a step more than match -- but not a significant issue

Reply:
The style I follow is to indent each `case` when the `match` uses curly braces, but to align each `case` with the `match` if the `match` does not use curly braces. I think that's what the Dafny style guide says (or in any case what I think it should say :)).